modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C

```diff
@@ -31,15 +29,22 @@ StressDivergenceTensors::StressDivergenceTensors(const std::string & name, Input
     _stress(getMaterialPropertyByName<RankTwoTensor>(_base_name + "stress")),
     _Jacobian_mult(getMaterialPropertyByName<ElasticityTensorR4>(_base_name + "Jacobian_mult")),
     _component(getParam<unsigned int>("component")),
-    _xdisp_coupled(isCoupled("disp_x")),
-    _ydisp_coupled(isCoupled("disp_y")),
-    _zdisp_coupled(isCoupled("disp_z")),
+    _ndisp(coupledComponents("displacements")),
+    _disp(3),
+    _xdisp_coupled(_ndisp >= 1),
+    _ydisp_coupled(_ndisp >= 2),
+    _zdisp_coupled(_ndisp >= 3),
     _temp_coupled(isCoupled("temp")),
-    _xdisp_var(_xdisp_coupled ? coupled("disp_x") : 0),
-    _ydisp_var(_ydisp_coupled ? coupled("disp_y") : 0),
-    _zdisp_var(_zdisp_coupled ? coupled("disp_z") : 0),
+    _xdisp_var(_xdisp_coupled ? coupled("displacements", 0) : 0),
+    _ydisp_var(_ydisp_coupled ? coupled("displacements", 1) : 0),
+    _zdisp_var(_zdisp_coupled ? coupled("displacements", 2) : 0),
     _temp_var(_temp_coupled ? coupled("temp") : 0)
 {
+  for (unsigned int i = 0; i < _ndisp; ++i)
+  {
+    _disp[i] = &coupledValue("displacements", i);
+
```

Comment:
Put ```_disp_var[i] = coupled("displacements, i);``` here. Otherwise we do not use {} blocks for single statements and... empty line.